compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1565,8 +1565,12 @@ private boolean isSelectivelyImmutableType(BType input, boolean disallowReadOnly
                 return isInherentlyImmutableType(tableConstraintType) ||
                         isSelectivelyImmutableType(tableConstraintType, unresolvedTypes, forceCheck, packageID);
             case TypeTags.UNION:
+                BUnionType unionType = (BUnionType) type;
+                if (unionType.getMemberTypes().size() == 0) {
```

Comment:
When is this empty? Is it for cyclic types? Are the member types checked later? Because there can be cyclic types that are not selectively immutable.

Reply:
It is for cyclic recursive types.
```
type A readonly & (A|B);
type A [T];
type B [T];
```

This change is apply to `A|B` in type A definition.